### PR TITLE
[EXE-1940] Adding `AiqDayDiff` PushDown to Snowflake Connector

### PR DESCRIPTION
**Pull request by yannistze**
Adding the Snowflake Connector part for the Flame PR https://github.com/ActionIQ/flame/pull/508 that introduced `AiqDayDiff`

Equivalent BigQuery PR: https://github.com/ActionIQ/spark-bigquery-connector/pull/12

## Test Notes
```
export SNOWFLAKE_TEST_ACCOUNT="aws"
export SKIP_BIG_DATA_TEST=true
export SPARK_LOCAL_IP=127.0.0.1
export SPARK_CONN_ENV_SFURL=xxx
export SPARK_CONN_ENV_SFUSER=xxx
export SPARK_CONN_ENV_SFPASSWORD=xxx
export SPARK_CONN_ENV_SFWAREHOUSE=DEMO_WH
export SPARK_CONN_ENV_SFDATABASE=TEST_DB
export SPARK_CONN_ENV_SFSCHEMA=PUBLIC
export SPARK_CONN_ENV_DBTABLE=SPARK_SNOWFLAKE_CI
```

```
sbt "it:testOnly net.snowflake.spark.snowflake.PushdownEnhancement01" >> /Users/yannis.tzemos/D
ocuments/CodeSnips/snow_test_output_1.txt
...
[info] Run completed in 1 minute, 6 seconds.
[info] Total number of tests run: 20
[info] Suites: completed 1, aborted 0
[info] Tests: succeeded 20, failed 0, canceled 0, ignored 0, pending 0
[info] All tests passed.
[success] Total time: 70 s (01:10), completed Jan 10, 2024, 1:12:25 AM
```

```
sbt "it:testOnly net.snowflake.spark.snowflake.PushdownEnhancement02" >> /Users/yannis.tzemos/Documents/CodeSnips/snow_test_output_2.txt
...
[info] - test pushdown number functions: PI() and Round()/Random
[info] - AIQ test pushdown day_start
[info] - AIQ test pushdown day_diff
[info] - AIQ test pushdown instr
[info] - AIQ test pushdown string_to_date
[info] - AIQ test pushdown date_to_string
[info] - test pushdown functions date_add/date_sub
[info] - test pushdown functions date_add/date_sub on Feb last day in leap and non-leap year
[info] - test pushdown WindowExpression: Rank without PARTITION BY
[info] - test pushdown WindowExpression: Rank with PARTITION BY
[info] - test pushdown WindowExpression: DenseRank without PARTITION BY
[info] - test pushdown WindowExpression: DenseRank with PARTITION BY
[info] - literal Date/Timestamp
[info] - test literal null
[info] - AIQ test approx_count_dist
[info] Run completed in 1 minute, 35 seconds.
[info] Total number of tests run: 17
[info] Suites: completed 1, aborted 0
[info] Tests: succeeded 17, failed 0, canceled 0, ignored 0, pending 0
[info] All tests passed.
[success] Total time: 97 s (01:37), completed Jan 10, 2024, 1:14:32 AM
```

## Deploy Notes
- `sbt publish` 
- Pick up the new Connector version in Flame

### CODE REVIEW

**Review by MasterDDT — APPROVED** [review]

File: src/it/scala/net/snowflake/spark/snowflake/PushdownEnhancement02.scala [hunk 1]
```diff
@@ -290,6 +290,54 @@ class PushdownEnhancement02 extends IntegrationSuiteBase {
     )
   }
 
+  test("AIQ test pushdown day_diff") {
+    jdbcUpdate(s"create or replace table $test_table_date " +
+      s"(startMs bigint, endMs bigint, tz string)")
+    jdbcUpdate(s"insert into $test_table_date values " +
+      s"""
+        |(1693609200000, 1693616400000, 'UTC'),
+        |(1693609200000, 1693616400000, 'EST')
```
Comment (MasterDDT) [inline src/it/scala/net/snowflake/spark/snowflake/PushdownEnhancement02.scala:299]: Also add some test cases with null?

Comment (yannistze) [reply src/it/scala/net/snowflake/spark/snowflake/PushdownEnhancement02.scala]: Done 👍

**Review by MasterDDT — APPROVED** [review]
